cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/IndexedObjectStore.java

```diff
@@ -140,6 +145,7 @@ public T read(String key) {
    * @param object object to be stored
    * @param secondaryKeys indices that can be used to lookup the object
    */
+  @WriteOnly
```

Comment:
Interesting, this seems like a write-only, but in order to maintain the index it actually involves a read (of the old value). So will the enforcement fail then?